### PR TITLE
Add 2.2.1

### PR DESCRIPTION
**Pull request by lilyminium**
This PR adds 2.2.1 as a force field to OpenFF force fields.

Working repo: https://github.com/openforcefield/sage-2.2.1 and release: https://github.com/openforcefield/sage-2.2.1/releases/tag/2.2.1

**Release checklist**
- [x] Open a new branch or fork on this repo. 
- [x] Add the new force field file to the branch. 
- [x] Update the date in the FF files
- [x] Update the authors in the `.zenodo.json` file - Details and validation info can be found [here](https://developers.zenodo.org/#add-metadata-to-your-github-repository-release)
	No new authors. IAlibay was added earlier.
- [x] Check that regular and `unconstrained` forcefields have constraints applied and absent, respectively
- [x] Ensure that there are no cosmetic attributes
- [x] Visually inspect for whitespace irregularities, etc. (Importantly: Tabs should be turned into 4 spaces)
- [x] Ensure that comments and authors are up to date
	No comments as the a27 angle wasn't manually adjusted. Authors are OpenFF
- [x] Ensure that `Bonds` `version="0.4"`, `fractional_bondorder_method="AM1-Wiberg"` and `fractional_bondorder_interpolation="linear"`
- [x] Ensure that `ProperTorsions` `version="0.4"`, `fractional_bondorder_method="AM1-Wiberg"` and `fractional_bondorder_interpolation="linear"`
- [x] Ensure the Electrostatics `scale14` value is `0.8333333333` (10 sig figs). Change it by hand if more 3's need to be added. 
- [x] Add monoatomic ion charges to the FF, just before the `ToolkitAM1BCC` tag
- [x] Ensure that new FFs are loadable in newest stable release of OFF toolkit
	Checked with 0.16.1
- [x] Draft a "version" description at the bottom of the README file in a branch or fork. _A zenodo DOI link is not necessary at this stage_
- [x] Prepare the fitting tarball (or other record of the files and/or code that produced the new parameters), which is often associated with a release like [this](https://github.com/openforcefield/openforcefield-forcebalance/releases/tag/v1.0.0-RC2). This collection of assets should include:
    - [x] a static copy of the data (QM and phys prop) used for the fit -- in repo itself
    - [x] a static copy of the scripts used for the fit -- in repo itself
    - [x] A file containing the environment used for the fit (like the output of `conda env export`) -- files are in repo
    - [x] Generally, anything else required to re-run the exact fit that was performed
- [x] Open a PR to master, for example [this](https://github.com/openforcefield/openforcefields/pull/6)

### CODE REVIEW

**Review by mattwthompson — APPROVED** [review]
Templates make this super easy

**Comment by mattwthompson** [comment]
I got a couple of canary failures after pushing changes that only touch comments/docs:

> [Canary tests on macOS-latest, Python 3.10](https://github.com/openforcefield/openff-forcefields/actions/runs/10814669914/job/30001680970?pr=113#logs)

```
__main__.HMRCanaryError: HMR tests failed:
[H][C]([H])=[C]([C]([H])([H])[H])[C]1([C]([H])([H])[O][N]([H])[H])[C]([H])([H])[C]1([H])[H]	openff-2.2.1	NaN position(s)
```

> [Canary tests on ubuntu-latest, Python 3.11](https://github.com/openforcefield/openff-forcefields/actions/runs/10814669914/job/30001681472?pr=113#logs)

```
HMRCanaryError: HMR tests failed:
[H][C]([H])=[C]([C]([H])([H])[H])[C]1([C]([H])([H])[O][N]([H])[H])[C]([H])([H])[C]1([H])[H]	openff-2.2.1	NaN position(s)
```

This is the molecule that failed:

image

It's notable that the same molecule failed in each case, and I ought to point out that two other elements in the matrix passed fine. I'm not sure what would have changed between a few hours ago and now. I'm re-kicking CI to see if this is intermittent or more genuine.

**Comment by mattwthompson** [comment]
Okay, re-running those tests came back green on that molecule (and all others). I'm bumping the CI matrix a little, from 4 to 6 canary jobs. If those all come back green I'm going to merge this as-is, if not I'll kick this out again for feedback.

File: canary/scripts/test_hmr.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from openff.toolkit.topology import Molecule
 from openmmforcefields.generators import SystemGenerator
 
-DATA_PATH = importlib.resources.files("canary") / "data"
+DATA_PATH = pathlib.Path(".") / "canary" / "data"
```
Comment (mattwthompson) [inline canary/scripts/test_hmr.py:10]: I'm surprised this ever worked; there was never a `canary/__init__.py` file or anything that could be used to install it.

**Comment by mattwthompson** [comment]
Okay, the same molecule failed again. I'm re-kick the canary tests again, which I'd expect to pass.

Still, I'd like input from @lilyminium before moving this forward - thumbs up/thumbs down on moving ahead with this? I'm inclined to go ahead, but am not comfortable being the sole decider here.

**Comment by lilyminium** [comment]
Thanks for looking into that @mattwthompson! I'm not sure either how concerned about this to be or where the concern should be aimed (e.g. if 2.2.0 might have intermittent issues as well). It does look like the "NaN position(s)" error is printed out for *any* exceptions raised that aren't NaN energies, so I'm going to kick CI a few times to see if I can get this to re-trigger and print the actual error.

**Comment by mattwthompson** [comment]
Nice finds - we're down to 0/12 failures if minimizing before running the canary tests, which sure seems like it fixes it. I'm re-kicking the tests a few times to be conservatie.

IIUC this doesn't invalidate the spirit of the test, which is to see if these molecules can run with HMR without crashing. It's just that that this molecule might have a bad conformer from RDKit a few percent of the time, and minimizing it against Sage fixes this and avoids bad initial coordinates for the OpenMM steps? (Assuming that NaN was the error, I haven't actually seen the real error in logs since you fixed that capturing.)

**Comment by lilyminium** [comment]
Just as an update -- [I kicked CI five times to get one error to pop up, still with the same molecule](https://github.com/openforcefield/openff-forcefields/actions/runs/10822111874). The message was indeed "NaN positions", but it's so intermittent that I wondered if the starting conformer might just be tricky. With the minimization code I ran CI 10 times and [they all passed](https://github.com/openforcefield/openff-forcefields/actions/runs/10824011524), as you probably saw when you ran it again!

**Comment by mattwthompson** [comment]
This is all to say that, if these tests continue to pass, I'm back to approving this PR. I interpret your comment to not be concerned about this, and also not be a blocker again yourself.

**Comment by lilyminium** [comment]
> IIUC this doesn't invalidate the spirit of the test, which is to see if these molecules can run with HMR without crashing. It's just that that this molecule might have a bad conformer from RDKit a few percent of the time, and minimizing it against Sage fixes this and avoids bad initial coordinates for the OpenMM steps? (Assuming that NaN was the error, I haven't actually seen the real error in logs since you fixed that capturing.)

Yes that's exactly what I'm thinking, although I couldn't reproduce this with 100 repeats on any local devices (probably because RDKit just generates the same conformer each time for me afaict).

**Comment by mattwthompson** [comment]
Everything's green, firing this off now

**Pull request merged**